docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,130 @@
+# Syncing LDAP Groups to OpenShift Groups
+
+OpenShift cluster admins need a way to sync their OpenShift `Groups` to external records in an LDAP server. This requires that we have a way of retrieving the LDAP users that are the members of a group, determining the OpenShift `Group` name for this collection, and updating the OpenShift record, removing the record as necessary when the corresponding LDAP record is removed. Necessary information about the LDAP server set-up and schema will be provided by the user in a `yaml` configuration file and referenced in the `oadm sync-groups` command.
+
+## Use Cases
+1. I want to sync all LDAP groups minus some blacklist to OpenShift `Groups`
+* I want to sync a specific set of LDAP groups to OpenShift `Groups`
+* I want to sync all OpenShift groups minus some blacklist with the LDAP server
+* I want to sync specific OpenShift `Groups` with the LDAP server records
+* I want to remove OpenShift `Groups` that were previously backed by LDAP records that no longer exist
+* I want to specify an explicit mapping from LDAP group identifiers to OpenShift `Group` resources
+
+## Supported LDAP Group Schemas
+1. groups as first-class LDAP entries that contain lists of members of the group
```

Comment:
Two main schemas. Member or group info can be in multiple attributes (e.g. `memberof,primaryGroupID` or `member,uniquemember`). Attribute value can be a DN or other unique attribute (e.g. `uid` or `gid`). What do you think of this:

1. Groups as LDAP entries with attribute(s) containing lists of members
  1. Member attribute value is the distinguished name (DN) of the corresponding member LDAP entry
  2. Member attribute value is a uniquely identifying attribute (e.g. uid) of the corresponding member LDAP entry
2. Users as LDAP entries with attribute(s) containing lists of groups they are a member of
  1. Group attribute value is a simple unique group name (no additional group metadata exists)
  2. Group attribute value is the DN of an LDAP entry containing additional group metadata
  3. Group attribute value is a uniquely identifying attribute (e.g. `gid`) of an LDAP entry containing additional group metadata